File: Programmers/Lv1/Lv1_2016년.py
def solution(a, b):
    answer = ''
    v = ["SUN","MON","TUE","WED","THU","FRI","SAT"]
    index = 0
    if a == 1 or a == 4 or a == 7:
        index = 5
    elif a == 2 or a == 8:
        index = 1
    elif a == 3 or a == 11:
        index = 2
    elif a == 5:
        index = 0
    elif a == 6:
        index = 3
    elif a == 9 or a == 12:
        index = 4
    elif a == 10:
        index = 6

    index += b % 7;
    if index > 6:
        index -= 7

    if index == 0:
        index = 6
    else:
        index -= 1
    return v[index]

print(solution(5,24))

File: Programmers/Lv1/Lv1_2016년.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

string solution(int a, int b) {
	vector<string> v = { "SUN","MON","TUE","WED","THU","FRI","SAT" };
	int index = 0;
	if (a == 1 || a==4 || a==7) index = 5;
	else if (a == 2 || a==8) index = 1;
	else if (a == 3 || a==11) index = 2;
	else if (a == 5) index = 0;
	else if (a == 6) index = 3;
	else if (a == 9 || a==12) index = 4;
	else if (a == 10) index = 6;

	index += b % 7;
	if (index > 6) index -= 7;

	if (index == 0)
		index = 6;
	else
		index--;
	return v[index];
}

